src/java/org/apache/cassandra/db/compaction/Verifier.java

```diff
@@ -361,12 +361,26 @@ public RangeOwnHelper(List<Range<Token>> normalizedRanges)
          * @throws RuntimeException if the key is not contained
          */
         public void check(DecoratedKey key)
+        {
+            if (!checkBoolean(key))
```

Comment:
Yep, this is exactly what I had in mind, thanks. Not sure of the names, though. Would prefer the throwing `check()` to be renamed into `validate()`, and `checkBoolean()` to become `check()`. `validate()` is the name we often give to such methods (that throw on failure).

Reply:
I am cherry picking this change. 

The only concern I have is we're now changing the signature and behavior of a pre-existing method. I generally try to avoid such changes to avoid confusing other authors.

Reply:
Please don't avoid it - unless it's in public API. Change as many layers as necessary, in general, until the new code fits into the big picture. We allow and welcome it here.

Reply:
Thanks, I will bear that in mind for future patches.